kafka/producer/sender.py

```diff
@@ -293,7 +296,9 @@ def _produce_request(self, node_id, acks, timeout, batches):
             produce_records_by_partition[topic][partition] = buf
 
         kwargs = {}
-        if self.config['api_version'] >= (0, 11):
+        if self.config['api_version'] >= (2, 1):
+            version = 7
```

Comment:
well, tbh i was king of hoping on second thoughts here, since i recently discovered that the 3 api's are related (broker, produce, fetch) but in what manner exactly, i could not say... this also applies to the other conditional. I realize there's a hole in the version checking logic there, but I'll probably just take it piece by piece, as you suggest, and do a bit more research as to what broker version plays nicely with what produce / fetch version

Reply:
poked into the console producer / consumers and came up with the proper pairings